lib/recorder.dart

```diff
@@ -1,131 +1,95 @@
-          ]),
-    );
+  void initAudioPlayer() {
+    audioPlayer = new AudioPlayer();
   }
 
-  _start() async {
+  void start() async {
     try {
       if (await AudioRecorder.hasPermissions) {
-          String path = "recorder.m4a";
-          Directory appDocDirectory = await getExternalStorageDirectory();
-          String filePath = appDocDirectory.path + '/' + path;
-          print("Start recording: $filePath");
-          File file = new File(filePath);
-          if(file.length() != null){
+        String path = "recorder.m4a";
+        Directory appDocDirectory = await getExternalStorageDirectory();
+        String filePath = appDocDirectory.path + '/' + path;
+        print("Start recording: $filePath");
+        File file = new File(filePath);
+        try {
+          if (file.exists() != null) {
             print("Deleted the fie in the path");
-              await file.delete();
+            await file.delete();
           }
-          await AudioRecorder.start(path: filePath, audioOutputFormat: AudioOutputFormat.AAC);
-        
+        } catch (e) {
+          print(e);
+        }
+
+        await AudioRecorder.start(
+            path: filePath, audioOutputFormat: AudioOutputFormat.AAC);
+
         bool isRecord = await AudioRecorder.isRecording;
-        setState(() {
-          _isRecorded = false;
-          _btnIcon = Icons.stop;
-          _btnClr = Colors.red;
-          _recording = new Recording(duration: new Duration(), path: "");
-          _isRecording = isRecord;
-        });
+        isRecorded = false;
+        recording = new Recording(duration: new Duration(), path: "");
+        isRecording = isRecord;
       } else {
-        Scaffold.of(context).showSnackBar(new SnackBar(content: new Text("You must accept permissions")));
+        print("Turn ON Permission inside Setting");
       }
-    } catch (e){
+    } catch (e) {
       print(e);
     }
   }
 
-  _stop() async {
+  void stop() async {
     var record = await AudioRecorder.stop();
     print("Stop recording: ${record.path}");
     bool isRecord = await AudioRecorder.isRecording;
     File file = new File(record.path);
     print(" File length: ${await file.length()}");
-    setState(() {
-      _isRecorded = true;
-      _btnIcon = Icons.mic;
-      _btnClr = Colors.white;
-      _recording = record;
-      _isRecording = isRecord;
-      _duration = record.duration;
-    });
-    _filePath = record.path;
+    isRecorded = true;
+    recording = record;
+    isRecording = isRecord;
+    duration = record.duration;
+    filePath = record.path;
   }
 
-  Future _playaudio() async {
-    await _audioPlayer.play(_filePath, isLocal: true);
-    setState(() {
-      _isRecorded = false;
-      _playMusicColor = Colors.red;
-      _playMusic = Icons.stop;
-      _playerState = PlayerState.playing;
-      });
-      new Future.delayed(_duration, () {
-              _stopaudio();
-            });
+  Future playaudio() async {
```

Comment:
Try to adhere to naming standards. If two words are joined, use cameCase. This should be playAudio